src/redis-cli.c

```diff
@@ -5785,6 +5804,36 @@ static dict *clusterManagerGetLinkStatus(void) {
     return status;
 }
 
+/* Check for nodes running in WSL without the cluster-announce-ip
+ * configuration. It returns a list of those nodes. */
+static list *clusterManagerGetMissingConfigNodes(void) {
+    if (cluster_manager.nodes == NULL) return NULL;
+    list *missingConfigNodes = listCreate();
+    if (!missingConfigNodes) return NULL;
+    listIter li;
+    listNode *ln;
+    listRewind(cluster_manager.nodes, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        clusterManagerNode *node = ln->value;
+        char *e = NULL;
+        redisReply *info = clusterManagerGetNodeRedisInfo(node, &e);
+        if (!info) {
+            if (e) zfree(e);
+            continue;
+        }
+        if (info->str) {
+            char *os = getInfoField(info->str, "os");
+            char *cluster_announce_ip = getInfoField(info->str,
+                                                     "cluster_announce_ip");
+            if (!os || !cluster_announce_ip) continue;
+            if ((strstr(os,"WSL") != NULL) && !strcmp(cluster_announce_ip, "0")) {
+                listAddNodeTail(missingConfigNodes, node);
```

Comment:
I wonder if it makes more sense to add this server side like how "ARM64-COW-BUG" works. Then it could self report if the config is not set. I'm not sure how much it makes sense in one tool to report a known issue.